jcl/src/java.base/share/classes/openj9/tools/attach/diagnostics/base/DiagnosticUtils.java

```diff
@@ -207,7 +228,35 @@ private static DiagnosticProperties getThreadInfo(String diagnosticCommand) {
 				bufferPrinter.println();
 			}
 			bufferPrinter.flush();
-			result = DiagnosticProperties.makeStringResult(buffer.toString());
+			result = makeStringResult(buffer.toString());
+		}
+		return result;
+	}
+
+	private static DiagnosticProperties doDump(String diagnosticCommand) {
+		DiagnosticProperties result = null;
+		String[] parts = diagnosticCommand.split(DIAGNOSTICS_OPTION_SEPARATOR);
+		logMessage("doDump: ", diagnosticCommand); //$NON-NLS-1$
+		if (parts.length != 2) {
+			result =  makeErrorProperties("Error: wrong number of arguments"); //$NON-NLS-1$
+		} else {
+			String dumpType;
+			/* handle legacy form of Dump.heap for compatibility with reference implementation */
+			if (DIAGNOSTICS_GC_HEAP_DUMP.equals(parts[0])) {
+				dumpType = "heap"; //$NON-NLS-1$
+			} else {
+				dumpType = parts[0].split("\\.")[1]; //$NON-NLS-1$
```

Comment:
How can this be sure that `parts[0]` contains any `'.'` characters?

Reply:
True.  Needs some error checking...